tests/network/vmi_istio.go

```diff
@@ -0,0 +1,297 @@
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		Describe("Inbound traffic", func() {
+			table.DescribeTable("request to VMI", func(port int) {
+				var job *batchv1.Job
+				job, err = verifyClientServerConnectivity(vmi, port)
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry(fmt.Sprintf("should reach VMI HTTP server on service declared port %d", svcDeclaredTestPort), svcDeclaredTestPort),
+				table.Entry(fmt.Sprintf("should reach VMI HTTP server on service undeclared port %d", svcUndeclaredTestPort), svcUndeclaredTestPort),
+			)
+			Context("With PeerAuthentication enforcing mTLS", func() {
+				var strictPeerAuthentication *istiogosecv1beta1.PeerAuthentication
+				BeforeEach(func() {
+					strictPeerAuthentication = &istiogosecv1beta1.PeerAuthentication{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "default",
+							Namespace: tests.NamespaceTestDefault,
+						},
+						Spec: istiosecv1beta1.PeerAuthentication{
+							Mtls: &istiosecv1beta1.PeerAuthentication_MutualTLS{
+								Mode: istiosecv1beta1.PeerAuthentication_MutualTLS_STRICT,
+							},
+						},
+					}
+
+					virtClient, err = kubecli.GetKubevirtClient()
+					tests.PanicOnError(err)
+
+					_, err = virtClient.IstioClient().SecurityV1beta1().PeerAuthentications(tests.NamespaceTestDefault).Create(context.TODO(), strictPeerAuthentication, metav1.CreateOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				AfterEach(func() {
+					err = virtClient.IstioClient().SecurityV1beta1().PeerAuthentications(tests.NamespaceTestDefault).Delete(context.TODO(), strictPeerAuthentication.Name, metav1.DeleteOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				table.DescribeTable("request to VMI", func(port int) {
+					var job *batchv1.Job
+					job, err = verifyClientServerConnectivity(vmi, port)
+
+					By("Waiting for the job to succeed")
+					Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).NotTo(Succeed())
+				},
+					table.Entry(fmt.Sprintf("client outside mesh should not reach VMI HTTP server on service declared port %d", svcDeclaredTestPort), svcDeclaredTestPort),
+					table.Entry(fmt.Sprintf("client outside mesh should not reach VMI HTTP server on service undeclared port %d", svcUndeclaredTestPort), svcUndeclaredTestPort),
+				)
+			})
+		})
+		Describe("Outbound traffic", func() {
+			It("Should be able to reach kubevirt.io", func() {
+				// Istio envoy may intercept the request, we need to check the header response, not just the
+				// return code of curl, because if envoy proxy forbids a request, it responds with 502 Bad Gateway code
+				err = console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: "curl -sD - -o /dev/null http://kubevirt.io | head -n 1\n"},
+					&expect.BExp{R: fmt.Sprintf(".*%s.*\n", "HTTP/1.1 301 Moved Permanently")},
```

Comment:
Are you sure this will always return this error code ? 

I just want to prevent us having to update the test if it starts returning 300 or 200 (or whatever other correct status code).

Reply:
You could extract it into a function `generatedExpectedHTTPStatusCodeString` or something alike.

Reply:
Done